magenta/musescore/magenta.qml

```diff
@@ -0,0 +1,109 @@
+import QtQuick 2.0
+import MuseScore 1.0
+
+MuseScore {
+      menuPath: "Plugins.Magenta"
+
+      function createNoteSequence() {
+        var noteSequence = {
+          timeSignatures: [],
+          keySignatures: [],
+          tempos: [],
+          notes: [],
+          totalTime: curScore.duration,
+          sourceType: 'SCORE',
+          ticksPerBeat: 220,
+        };
+
+        var cursor = curScore.newCursor();
+        var startStaff;
+        var endStaff;
+        var endTick;
+        var fullScore = false;
+        cursor.rewind(1);
+        if (!cursor.segment) { // no selection
+           fullScore = true;
+           startStaff = 0; // start with 1st staff
+           endStaff = curScore.nstaves - 1; // and end with last
+           endTick = curScore.lastSegment.tick + 1;
+        } else {
+           startStaff = cursor.staffIdx;
+           cursor.rewind(2);
+           if (cursor.tick == 0) {
+             // this happens when the selection includes
+             // the last measure of the score.
+             // rewind(2) goes behind the last segment (where
+             // there's none) and sets tick=0
+             endTick = curScore.lastSegment.tick + 1;
+           } else {
+             endTick = cursor.tick;
+           }
+           endStaff = cursor.staffIdx;
```

Comment:
What happens when multiple staves are selected?  Does cursor.rewind(2) go to the bottom of the staff or has a default staff it would go to?  I only remember in time it would rewind to the end of piece (I don't have a mac with me to check it out MuseScore, and might be confusing between rewind(1) or rewind(2).